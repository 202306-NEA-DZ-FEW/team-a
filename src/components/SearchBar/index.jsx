import { useRouter } from "next/router";
import { useEffect, useState } from "react";
import { RiSearch2Line } from "react-icons/ri";
import { useDebounce } from "use-debounce";

function SearchBar({ t, queryParams }) {
  const router = useRouter();
  const [text, setText] = useState("");
  let [query] = useDebounce(text, 600);

  useEffect(() => {
    // Update the query state when text or queryParams change
    const newQueryParams = { ...queryParams };
    if (query) {
      newQueryParams.search = query;
    } else {
      delete newQueryParams.search;
    }

    const currentURL = router.query.search || "";

    // Only update the URL if it's different from the current URL
    if (query !== currentURL) {
      router.push(
        {
          pathname: `/products`,
          query: newQueryParams,
        },
        undefined,
        { scroll: false }
      );
    }
  }, [query, queryParams, router]);

  useEffect(() => {
    // Update the text input when queryParams change
    setText(queryParams.search || "");
  }, [queryParams.search]);

  return (
    <div className='relative rounded-full border'>
      <RiSearch2Line className='absolute right-3 translate-y-1/2 transform text-gray-500' />
      <input
        value={text}
        onChange={(e) => setText(e.target.value)}
        type='text'
        placeholder={t("productsPage:searchPlaceHolder")}
        className='border-none input-sm rounded-full input-bordered input bg-gray-50 outline-none font-light normal-case w-full'
      />
    </div>
  );
}

export default SearchBar;
