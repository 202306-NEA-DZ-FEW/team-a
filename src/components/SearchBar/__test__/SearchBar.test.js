import renderer from "react-test-renderer";

import SearchBar from "..";

it("renders correctly", () => {
  const queryParams = {};
  const mockT = jest.fn();
  const tree = renderer
    .create(<SearchBar queryParams={queryParams} t={mockT} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
