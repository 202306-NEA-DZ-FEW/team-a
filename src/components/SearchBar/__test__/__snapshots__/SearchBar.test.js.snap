// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="relative rounded-full border"
>
  <svg
    className="absolute right-3 translate-y-1/2 transform text-gray-500"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M11 2C15.968 2 20 6.032 20 11C20 15.968 15.968 20 11 20C6.032 20 2 15.968 2 11C2 6.032 6.032 2 11 2ZM11 18C14.8675 18 18 14.8675 18 11C18 7.1325 14.8675 4 11 4C7.1325 4 4 7.1325 4 11C4 14.8675 7.1325 18 11 18ZM19.4853 18.0711L22.3137 20.8995L20.8995 22.3137L18.0711 19.4853L19.4853 18.0711Z"
    />
  </svg>
  <input
    className="border-none input-sm rounded-full input-bordered input bg-gray-50 outline-none font-light normal-case w-full"
    onChange={[Function]}
    type="text"
    value=""
  />
</div>
`;
